pyo3-build-config/src/impl_.rs

```diff
@@ -1651,14 +1651,19 @@ pub fn make_cross_compile_config() -> Result<Option<InterpreterConfig>> {
 /// Only used by `pyo3-build-config` build script.
 #[allow(dead_code)]
 pub fn make_interpreter_config() -> Result<InterpreterConfig> {
+    let host = Triple::host();
     let abi3_version = get_abi3_version();
 
-    if have_python_interpreter() {
+    // See if we can skip the Python interpreter configuration detection.
+    // Unix "abi3" extension modules can be built without any interpreter,
+    // so do this by default just like maturin does.
+    let require_interpreter = abi3_version.is_none() || is_linking_libpython_for_target(&host);
```

Comment:
I think we should also check `python-dll-a` feature on Windows once #2282 lands?

Reply:
Yes, that should also work. It's draft, and if there's a consensus on the related issue, I'll implement it too.

This will also require updating the user-facing documentation in many places.